server/plugins/schema-plugin-ob-oracle/src/main/java/com/oceanbase/odc/plugin/schema/oboracle/parser/OBOracleGetDBTableByParser.java

```diff
@@ -470,32 +469,37 @@ private void fillSubPartitions(DBTablePartition partition, Partition partitionSt
                 .setPartitionsNum(
                         parsedSubPartitionOption.getTemplates() != null ? parsedSubPartitionOption.getTemplates().size()
                                 : null);
-        for (PartitionElement partitionElement : partitionStmt.getPartitionElements()) {
-            if (partitionElement.getSubPartitionElements() != null) {
+        for (int i = 0; i < partitionStmt.getPartitionElements().size(); i++) {
+            DBTablePartitionDefinition partitionDefinition = partition.getPartitionDefinitions().get(i);
+            if (partitionStmt.getPartitionElements().get(i).getSubPartitionElements() != null) {
                 // obtain DBTablePartitionDefinitions for non-templated subpartitions
-                for (int i = 0; i < partitionElement.getSubPartitionElements().size(); i++) {
+                for (int j = 0; j < partitionStmt.getPartitionElements().get(i).getSubPartitionElements().size(); j++) {
                     DBTablePartitionDefinition subPartitionDefinition = new DBTablePartitionDefinition();
-                    SubPartitionElement subPartitionElement = partitionElement.getSubPartitionElements().get(i);
+                    SubPartitionElement subPartitionElement =
```

Comment:
this logic is so close to code above is they have some logic can be used common?

Reply:
A default method will be added in the super interface `GetDBTableByParser`

Reply:
A default method will be added to the super interface `GetDBTableByParser`  to avoid duplicate code